File: web/js/EditUser.js
function editUser(){
    var phoneNum = $("#phoneNum").val();
    var userId = $("#userId").val();
    var shortNum = $("#shortNum").val();
    var mobile = $("#mobile").val();
    var time = $("#time").val();
    var ipadd = $("#ipadd").val();
    var vlan = $("#vlan").val();
    var tactics = $("#tactics").val();
    var status = $("#status").val();
    var balance = $("#balance").val();
    var stored = $("#stored").val();
    var mtime = $("#mTime").val();
    var startTime = $("#time").val();
    var protocal = $("#protocal").val();
    var id = $("#id").val();
    var gate = $("#gate").val();
    var tbl = $("#tbl").val();

    var type='update'
    $.post("EditUser",{type:type,id:id,phoneNum:phoneNum,userId:userId,shortNum:shortNum,mobile:mobile,
    time:time,ipadd:ipadd,vlan:vlan,tactics:tactics,status:status,balance:balance,stored:stored,mtime:mtime,
        time:startTime,protocal:protocal,gate:gate,tbl:tbl},function(data){
        if(data=='success'){
            alert('修改成功！');
            window.location.href='ViewUser.jsp';
        }
        else{
            alert('修改失败，请联系管理员。');
        }
    })
}


function initUserInfo(){
    var phoneNum = $("#phoneNum").val();
    var userId = $("#userId").val();
    var shortNum = $("#shortNum").val();
    var mobile = $("#mobile").val();
    var time = $("#time").val();
    var ipadd = $("#ipadd").val();
    var vlan = $("#vlan").val();
    var tactics = $("#tactics").val();
    var status = $("#status").val();
    var balance = $("#balance").val();
    var stored = $("#stored").val();
    var mtime = $("#mTime").val();
    var startTime = $("#time").val();
    var protocal = $("#protocal").val();
    var id = $("#id").val();
    var gate = $("#gate").val();
    var tbl = $("#tbl").val();
    var type='delete';
    var tbl = $("#tbl").val();
    if(confirm('确定要删除此用户吗？\n\r注意：删除此用户后这个号码会返回未使用号码池，并在设备上解除注册。')){
        $("#btn_save").attr("disabled","true");
        $("#btn_del").attr("disabled","true");
        $("#btn_back").attr("disabled","true");
        $("#btn_openLocal").attr("disabled","true");
        $("#btn_closeLocal").attr("disabled","true");
        $("#btn_openLong").attr("disabled","true");
        $("#btn_closeLong").attr("disabled","true");
        $.post("EditUser",{type:type,id:id,phoneNum:phoneNum,userId:userId,shortNum:shortNum,mobile:mobile,
            time:time,ipadd:ipadd,vlan:vlan,tactics:tactics,status:status,balance:balance,stored:stored,mtime:mtime,time:startTime,protocal:protocal,tbl:tbl},function(data){
            if(data == 'success'){
                window.location.href='uSuccess.jsp';
            }
            else if(data == 'fail'){
                alert('删除失败！请重试...');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
        });
    }
    else{
        return;
    }
}

function openLocal(){
    var shortNum = $("#shortNum").val();
    var type = 'openLocal';
    if(confirm('要给'+shortNum+'号码开通本地通话业务吗？')){
        $("#btn_save").attr("disabled","true");
        $("#btn_del").attr("disabled","true");
        $("#btn_back").attr("disabled","true");
        $("#btn_openLocal").attr("disabled","true");
        $("#btn_closeLocal").attr("disabled","true");
        $("#btn_openLong").attr("disabled","true");
        $("#btn_closeLong").attr("disabled","true");
        $.post("EditUser",{shortNum:shortNum,type:type},function(data){
            if(data=='openLocalSuccess'){
                alert('号码本地通话业务开通成功！');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
            else{
                alert('开通失败！该号码可能已经开通此项业务。');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
        });
    }
    else{
        return;
    }
}

function offLocal(){
    var shortNum = $("#shortNum").val();
    var type='offLocal'
    if(confirm('要给'+shortNum+'号码关闭本地通话业务吗？')){
        $("#btn_save").attr("disabled","true");
        $("#btn_del").attr("disabled","true");
        $("#btn_back").attr("disabled","true");
        $("#btn_openLocal").attr("disabled","true");
        $("#btn_closeLocal").attr("disabled","true");
        $("#btn_openLong").attr("disabled","true");
        $("#btn_closeLong").attr("disabled","true");
        $.post("EditUser",{shortNum:shortNum,type:type},function(data){
            if(data=='offLocalSuccess'){
                alert('号码本地通话业务关闭成功！');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
            else{
                alert('关闭失败！请重试。');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
        });
    }
    else{
        return;
    }
}

function openLong(){
    var shortNum = $("#shortNum").val();
    var type='openLong';
    if(confirm('要给'+shortNum+'号码开通长途通话业务吗？')){
        $("#btn_save").attr("disabled","true");
        $("#btn_del").attr("disabled","true");
        $("#btn_back").attr("disabled","true");
        $("#btn_openLocal").attr("disabled","true");
        $("#btn_closeLocal").attr("disabled","true");
        $("#btn_openLong").attr("disabled","true");
        $("#btn_closeLong").attr("disabled","true");
        $.post("EditUser",{shortNum:shortNum,type:type},function(data){
            if(data=='openLongSuccess'){
                alert('号码长途通话业务开通成功！');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
            else{
                alert('开通失败！该号码可能已经开通此项业务。');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
        });
    }
    else{
        return;
    }
}

function offLong(){
    var shortNum = $("#shortNum").val();
    var type = 'offLong';
    if(confirm('要给'+shortNum+'号码关闭长途通话业务吗？')){
        $("#btn_save").attr("disabled","true");
        $("#btn_del").attr("disabled","true");
        $("#btn_back").attr("disabled","true");
        $("#btn_openLocal").attr("disabled","true");
        $("#btn_closeLocal").attr("disabled","true");
        $("#btn_openLong").attr("disabled","true");
        $("#btn_closeLong").attr("disabled","true");
        $.post("EditUser",{shortNum:shortNum,type:type},function(data){
            if(data=='offLongSuccess'){
                alert('号码长途通话业务关闭成功！');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
            else{
                alert('关闭失败！请重试。');
                $("#btn_save").removeAttr("disabled");
                $("#btn_del").removeAttr("disabled");
                $("#btn_back").removeAttr("disabled");
                $("#btn_openLocal").removeAttr("disabled");
                $("#btn_closeLocal").removeAttr("disabled");
                $("#btn_openLong").removeAttr("disabled");
                $("#btn_closeLong").removeAttr("disabled");
            }
        });
    }
    else{
        return;
    }
}

File: src/com/bill/action/EditUser.java
package com.bill.action;

import com.bill.dao.BillSysDAO;
import com.bill.dao.BillSysDAOImpl;
import com.bill.pojo.BillSys_User;
import com.bill.pojo.gtao_Phone_User;
import com.bill.tool.SshTool;
import com.opensymphony.xwork2.ActionSupport;
import org.apache.struts2.ServletActionContext;

import javax.servlet.http.HttpServletResponse;
import java.io.PrintWriter;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * User: HP
 * Date: 13-6-22
 * Time: 下午3:39
 * To change this template use File | Settings | File Templates.
 */
public class EditUser extends ActionSupport {
    private int id;
    private String phoneNum;
    private String userId;
    private String shortNum;
    private String mobile;
    private String time;
    private String ipadd;
    private String vlan;
    private String tactics;
    private String status;
    private String balance;
    private String stored;
    private String mtime;
    private String type;
    private String tbl;
    private String protocal;
    private String gate;
    private String opentime;
    private BillSysDAO billService;

    public BillSysDAO getBillService() {
        return billService;
    }

    public void setBillService(BillSysDAO billService) {
        this.billService = billService;
    }

    private List userGroupList = new ArrayList();

    public String getOpentime() {
        return opentime;
    }

    public void setOpentime(String opentime) {
        this.opentime = opentime;
    }

    public String getGate() {
        return gate;
    }

    public void setGate(String gate) {
        this.gate = gate;
    }

    public String getProtocal() {
        return protocal;
    }

    public void setProtocal(String protocal) {
        this.protocal = protocal;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getPhoneNum() {
        return phoneNum;
    }

    public void setPhoneNum(String phoneNum) {
        this.phoneNum = phoneNum;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getShortNum() {
        return shortNum;
    }

    public void setShortNum(String shortNum) {
        this.shortNum = shortNum;
    }

    public String getMobile() {
        return mobile;
    }

    public void setMobile(String mobile) {
        this.mobile = mobile;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getIpadd() {
        return ipadd;
    }

    public void setIpadd(String ipadd) {
        this.ipadd = ipadd;
    }

    public String getVlan() {
        return vlan;
    }

    public void setVlan(String vlan) {
        this.vlan = vlan;
    }

    public String getTactics() {
        return tactics;
    }

    public void setTactics(String tactics) {
        this.tactics = tactics;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getTbl() {
        return tbl;
    }

    public void setTbl(String tbl) {
        this.tbl = tbl;
    }

    public String getBalance() {
        return balance;
    }

    public void setBalance(String balance) {
        this.balance = balance;
    }

    public String getStored() {
        return stored;
    }

    public void setStored(String stored) {
        this.stored = stored;
    }

    public String getMtime() {
        return mtime;
    }

    public void setMtime(String mtime) {
        this.mtime = mtime;
    }

    @Override
    public String execute() throws Exception {
        SshTool tool = new SshTool();
        HttpServletResponse response = ServletActionContext.getResponse();
        response.setContentType("text/html;charset=GBK");
        PrintWriter out = response.getWriter();
        /**
         * type==update,修改用户信息
         */
        if(type.equals("update")||type=="update"){
            gtao_Phone_User user = new gtao_Phone_User();
            user.setId(id);
            user.setLongNum(phoneNum);
            user.setShortNum(shortNum);
            user.setUserid(userId);
            user.setMobile(mobile);
            user.setItime(time);
            user.setLastUpd(this.getNow());
            user.setMaturityTime(mtime);
            user.setPhoneIp(ipadd);
            user.setVlan(vlan);
            user.setStatus(status);
            user.setBalance(balance);
            user.setTactics(tactics);
            user.setStored(stored);
            user.setProtocal(protocal);
            user.setTbl(tbl);
            user.setGate(gate);
            user.setEmail(opentime);
            boolean flag = billService.updateUserInfo(user);
            if(flag){
                out.print("success");
            }
            else {
                out.print("fail");
            }
            out.flush();
            out.close();
            //update end.
        }
        /**
         * 删除用户
         */
        if(type.equals("delete")||type=="delete"){
            BillSys_User userInfo = new BillSys_User();
            userInfo.setUsername(userId);
            userInfo.setPassword(mobile);
            userInfo.setLevel(2);
            boolean flag = false;
            gtao_Phone_User user = new gtao_Phone_User();
            user.setId(id);
            user.setLongNum(phoneNum);
            if(tbl.contains("sale")){
                if(tool.DelUser(phoneNum, shortNum, ipadd, protocal, userId)){
                    if(billService.delInfoFromUser(id,phoneNum)){
                        flag = billService.initSaleApplyDetail(phoneNum, userId, tbl);
                        //REMOVE USER
                    }
                }
            }
            else {
                if(tool.DelUser(phoneNum,shortNum,ipadd,protocal,userId)){
                    if(billService.delInfoFromUser(id,phoneNum)){
                        flag = billService.initApplyDetail(tbl, phoneNum, userId);
                        //REMOVE USER
                    }
                }
            }
            if(flag){
                out.print("success");
            }
            else{
                out.print("fail");
            }
            out.flush();
            out.close();
        }
        //开通市话
        if(type.equals("openLocal")){
            if(tool.userCallCompetenceManage(shortNum,type)){
                out.print("openLocalSuccess");
            }
            else {
                out.print("fail");
            }
        }
        //关闭市话
        if(type.equals("offLocal")){
            if(tool.userCallCompetenceManage(shortNum,type)){
                out.print("offLocalSuccess");
            }
            else {
                out.print("fail");
            }
        }
        //开通长途
        if(type.equals("openLong")){
            if(tool.userCallCompetenceManage(shortNum,type)){
                out.print("openLongSuccess");
            }
            else {
                out.print("fail");
            }
        }
        //关闭长途
        if(type.equals("offLong")){
            if(tool.userCallCompetenceManage(shortNum,type)){
                out.print("closeLongSuccess");
            }
            else {
                out.print("fail");
            }
        }
        out.flush();
        out.close();
        return null;
    }

    public String getNow(){
        Date date = new Date();
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        String time = sdf.format(date);
        return time;
    }

    public String getUserGroupList(){
        userGroupList = new BillSysDAOImpl().getAllGroup();
        return null;
    }
}
